src/lib/__tests__/chartStorage.js

```diff
@@ -0,0 +1,310 @@
+    window.localStorage.setItem("savedCharts", JSON.stringify(expectedData)); 
+    // Call the function to test. 
+    const result = loadAllSavedCharts(); 
+    // Check if the output is correct. 
+    expect(result).toEqual(expectedData);
+    window.localStorage.clear();
+  });
+  
+ // If 2+ chart is saved. 
+
+  test("Expect, if there is 2+ chart currently saved, loadAllSavedCharts to return all of it. ", () => {
+    window.localStorage.clear(); 
+    // Create the data. 
+    const expectedData = [
+        {
+          type: "line",
+          data: [
+            { x: 1, y: 10 },
+            { x: 2, y: 20 },
+            { x: 3, y: 30 },
+          ],
+          xLabel: "X Axis",
+          yLabel: "Y Axis",
+          title: "Sample Chart",
+          color: "#FFFFFF",
+        },
+        {
+          type: "line",
+          data: [
+            { x: 1, y: 10 },
+            { x: 3, y: 30 },
+            { x: 2, y: 20 },
+          ],
+          xLabel: "X Axis",
+          yLabel: "Y Axis",
+          title: "Sample Chart",
+          color: "#FFFFFF",
+        },
+    ]; 
+    // Create the setup. 
+    window.localStorage.setItem("savedCharts", JSON.stringify(expectedData)); 
+    // Call the function to test. 
+    const result = loadAllSavedCharts(); 
+    // Check if the output is correct. 
+    expect(result).toEqual(expectedData); 
+    window.localStorage.clear();
+  });
+  
+  // If 0 chart is saved. 
+
+  test("Expect, if there is 0 chart currently saved, loadAllSavedCharts to return an em,pty list. ", () => {
+    window.localStorage.clear();
+    window.localStorage.removeItem("savedCharts"); 
+    // Call the function to test. 
+    const result = loadAllSavedCharts(); 
+    // Check if the output is correct. 
+    expect(result).toEqual([]); 
+    window.localStorage.clear() ; 
+  }); 
+
+// Function 3. 
+
+// If 2 chart is saved, get the 2'nd chart. 
+
+test( "Expect, if 2 chart is saved, index input is 1, loadSavedChart to return the 2'nd chart. " , function () { 
+    window.localStorage.clear() ; 
+    // Create the data. 
+    const chart_1 = {
+        type: 'line',
+        data: [
+            { x: 1, y: 10 },
+            { x: 2, y: 20 },
+            { x: 3, y: 30 }
+        ],
+        xLabel: 'X Axis',
+        yLabel: 'Y Axis',
+        title: 'Sample Chart',
+        color: '#FFFFFF'
+    };
+    const chart_2 = {
+        type: 'line',
+        data: [
+            { x: 1, y: 10 },
+            { x: 3, y: 30 },
+            { x: 2, y: 20 }
+        ],
+        xLabel: 'X Axis',
+        yLabel: 'Y Axis',
+        title: 'Sample Chart',
+        color: '#FFFFFF'
+    }; 
+    chartList = [ chart_1 , chart_2 ] ; 
+    // Create the setup. 
+    window.localStorage.setItem("savedCharts", JSON.stringify(chartList)); 
+    // Call the function to test. 
+    // Check if the output is correct. 
+    expect( loadSavedChart( 1 ) ).toEqual( chart_2 ) ; 
+    window.localStorage.clear() ; 
+}) ; 
+
```

Comment:
Maybe add a test when the index it out of bounds? Or when the savedCharts has 0 items? Just to verify that undefined is returned, instead of a specific value. This is more of an implementation test, but might still be worth having.